File: Surface_tracking_matlab/Real2complex.m
function U = Real2complex(maxOrder)
% Contribution
%  Author: Xinyu Nie
%  Created: 2022/11/10
%  Copyright:The Neuro Image Computing Research (NICR) group at the Mark and Mary Stevens Neuroimaging 
%  and Informatics Institute of USC Laboratory of NeuroImaging 
%  USC Stevens Neuroimaging and Informatics Institute
%  email: xnie@usc.edu

N=(maxOrder+1)*(maxOrder+2)/2;

U=zeros(N,N);
U(1,1)=1;
t=1;
for order = 2:2:maxOrder
    len=2*order+1;
    A=zeros(len,len);
    A(order+1,order+1)=1;
    k1=sqrt(1/2);
    k2=i*k1;
    for m=1:order
         A(m+order+1,order+1-m)=k2*((-1)^(m));
         A(m+order+1,order+1+m)=k1*((-1)^(m));
          A(order+1-m,order+1-m)=-k2;
         A(order+1-m,order+1+m)=k1;
    end
    U(t+1:t+len,t+1:t+len)=A;
    t=t+len;
end

end

File: Surface_tracking_c++/SurfacePro/Real2complex.cpp
#include"Real2complex.h"

vector<vector<complex<double>>> Real2complex(int maxOrder){
    int N = (maxOrder+1)*(maxOrder+2)/2;
    vector<vector<complex<double>>> U(N,vector<complex<double>>(N,0));
    U[0][0] = 1;
    int t = 1;
    for (int order = 2; order<=maxOrder;order+=2){
        int len = order*2+1;
        vector<vector<complex<double>>> A(len, vector<complex<double>>(len, 0));
        A[order][order]=1;
        float k1=sqrt(0.5);
        complex<double> k2(0,k1);
        for (int m = 1;m<=order;m++){
            int flag;
            if (m%2) flag = -1;
            else flag = 1;
            A[m+order][order-m] = k2*complex<double>(flag);
            A[m+order][order+m] = complex<double>(k1)*complex<double>(flag);
            A[order-m][order-m] = -complex<double>(k2);
            A[order-m][order+m]= complex<double>(k1);
        }
        for (int i = 0;i<len;i++){
            for (int j = 0;j<len;j++){
                U[t+i][t+j] = A[i][j];
            }
        }
        t+=len;
    }
    return U;
}